File: model/client_model.py
from binascii import Error
from model import connection
from model import utils

class ClientModel(connection.Conection):
    def __init__(self):
        super().__init__()
    
    def agregar_cliente(self, datos):
        try:
            self.cursor.execute("""
                INSERT INTO clientes (id_persona) VALUES (%s)
            """, (datos["id_persona"],))
            self.conn.commit()
            print("Cliente agregado satisfactoriamente.")
        except Error as err:
            print(f"Error al agregar cliente: {err}")
            self.conn.rollback()

    def ver_clientes(self):
        try:
            self.cursor.execute("SELECT cl.id, pe.primer_nombre, pe.documento FROM clientes cl JOIN personas pe ON cl.id_persona = pe.id")
            return self.cursor.fetchall()
        except Error as err:
            print(f"Error al obtener clientes: {err}")
            return None

    def actualizar_campo(self, id_cliente, campo, nuevo_valor):
        try:
            query = f"UPDATE clientes SET {campo} = %s WHERE id = %s"
            self.cursor.execute(query, (nuevo_valor, id_cliente))
            self.conn.commit()
            print(f"Campo '{campo}' actualizado correctamente para el cliente con ID {id_cliente}.")
        except Error as err:
            print(f"Error al actualizar el campo: {err}")
            self.conn.rollback()

    def eliminar_cliente(self, id_cliente):
        try:
            query = "DELETE FROM clientes WHERE id = %s"
            self.cursor.execute(query, (id_cliente,))
            self.conn.commit()
            print(f"Cliente con ID {id_cliente} eliminado correctamente.")
        except Error as err:
            print(f"Error al eliminar cliente: {err}")
            self.conn.rollback()

    def validar_id_cliente(self, id_cliente):
        try:
            self.cursor.execute(
                "SELECT id FROM clientes WHERE id = %s", (id_cliente,)
            )
            return self.cursor.fetchone() is not None
        except Error as e:
            print(f"Error al validar el ID: {e}")
            return False

    def validar_id_persona(self, id_persona):
        try:
            self.cursor.execute(
                "SELECT id FROM personas WHERE id = %s", (id_persona,)
            )
            return self.cursor.fetchone() is not None
        except Error as e:
            print(f"Error al validar el ID de persona: {e}")
            return False

File: model/ClientModel.php
<?php
require_once 'Connection.php';

class ClientModel extends Connection {
    public function __construct() {
        parent::__construct();
    }

    public function agregarCliente($idPersona) {
        try {
            $stmt = $this->conn->prepare("INSERT INTO clientes (id_persona) VALUES (?)");
            $stmt->execute([$idPersona]);
            return true;
        } catch(PDOException $e) {
            echo "Error: " . $e->getMessage();
            return false;
        }
    }

    public function obtenerClientes() {
        try {
            $sql = "SELECT c.id, p.* FROM clientes c 
                    JOIN personas p ON c.id_persona = p.id";
            $stmt = $this->conn->query($sql);
            return $stmt->fetchAll(PDO::FETCH_ASSOC);
        } catch(PDOException $e) {
            echo "Error: " . $e->getMessage();
            return [];
        }
    }

    public function obtenerClientePorId($id) {
        try {
            $sql = "SELECT c.id, p.* FROM clientes c 
                    JOIN personas p ON c.id_persona = p.id 
                    WHERE c.id = ?";
            $stmt = $this->conn->prepare($sql);
            $stmt->execute([$id]);
            return $stmt->fetch(PDO::FETCH_ASSOC);
        } catch(PDOException $e) {
            echo "Error: " . $e->getMessage();
            return null;
        }
    }

    public function eliminarCliente($id) {
        try {
            $stmt = $this->conn->prepare("DELETE FROM clientes WHERE id = ?");
            $stmt->execute([$id]);
            return true;
        } catch(PDOException $e) {
            echo "Error: " . $e->getMessage();
            return false;
        }
    }
}
?>